### DISCUSSION TITLE
Need periodic MQTT updates for AC powered z-wave plug

### DISCUSSION THREAD

**Discussion by AlfaBravoX** [zwave-js/zwave-js-ui#3280] [Q&A]
Hi community,

my use case is to setup dashboard with mqtt controlled z-way elements (switches and thermostat). I need to mark different color of those elements that were not sending MQTT heart beat over  5 minutes. 

For frequently waking up battery powered devices it is not a big problem, because after wake status is changed and mqtt message is fired on, but the probem is  with AC powered z-wave devices ie. plugs. These are not sending regular updates.

Using `z-wave/_CLIENTS/ZWAVE_GATEWAY-office/api/driverFunction` and payload `'{"args":["const nodeId = 1; const node=this.zwaveClient._nodes.get(nodeId); return node.lastActive"]}'` is not good solution for me because I'd like to stick to this topic only: `z-wave/myNodeId/37/0/currentValue` 

Any ideas how to enable ZWJS UI to send periodic MQTT updates/heart beats on above mentioned topic would be appreciated

**Comment by robertsLando** [comment]
Cannot you use [lastActive](https://zwave-js.github.io/zwave-js-ui/#/guide/mqtt?id=node-last-active) topic?

Alternatively you could use [Scheduled jobs](https://zwave-js.github.io/zwave-js-ui/#/usage/setup?id=scheduled-jobs) in order to make zui programmatically send updates but I prefer the first solution

**Reply by AlfaBravoX** [reply]
Unfortunately `z-wave/myNodeId/lastActive` does not emit anything regularly. Currently timestamp is 2 hour late even thought UI last timestamp is just 30 mins old. So it is not even in sync with UI.

**Reply by robertsLando** [reply]
Strange, it is updated every time stats are emitted: https://github.com/zwave-js/zwave-js-ui/blob/e38b13fea8fb637f072771b31a71ec8e703964b7/lib/ZwaveClient.ts#L5429

@AlCalzone is lastSeen always set in stats?

**Reply by AlCalzone** [reply]
It should always be set when there's communication with the node. @AlfaBravoX please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs), on loglevel `debug`, set it to save to file.

When this happens again, share that log and the `<random-number>.jsonl` file from your store directory. Attach them here as a file (drag & drop into the text field).

**Comment by AlfaBravoX** [comment]
You can see that lastActive topic is 3 days old, however UI is uptodate:

<img width="483" alt="image" src="https://github.com/zwave-js/zwave-js-ui/assets/11148591/a2f88c5d-7b89-405e-9e3a-0f238bfa5b5b">

```
Client (null) received PUBLISH (d0, q0, r1, m0, 'chbk32-z-wave/myNodeId/lastActive', ... (44 bytes))
{"time":1694434125943,"value":1694434125596}
```
Logs attached:
[zwavejs_2023-09-14_lastActive.log](https://github.com/zwave-js/zwave-js-ui/files/12606625/zwavejs_2023-09-14_lastActive.log)
[jsonl_logs.zip](https://github.com/zwave-js/zwave-js-ui/files/12606651/jsonl_logs.zip)

**Reply by robertsLando** [reply]
cc @AlCalzone I don't see anything wrong here at first sight

**Reply by AlCalzone** [reply]
@AlfaBravoX I'll need some more context here:

- Which is your actual home ID? `d30c5c0f` or `de583a74`? Check the info icon in the top right corner to see.
- Which node ID is this example for?

**Comment by AlfaBravoX** [comment]
```
zwave-js-ui: 8.25.1.9b831e2
zwave-js: 11.14.2
home id: 3540802575
home hex: 0xd30c5c0f
```

does it help?

**Reply by AlCalzone** [reply]
> Which node ID is the above example for?

**Reply by AlfaBravoX** [reply]
The only node that is AC powered is node 6. The issue is there are no periodic MQTT updates for AC powered z-wave plug, node id 6.
